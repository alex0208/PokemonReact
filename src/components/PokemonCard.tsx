import {
    Button,
    capitalize,
    Card,
    CardActionArea,
    CardActions,
    CardContent,
    CardMedia,
    Typography,
} from '@mui/material';

import { Box, styled } from '@mui/system';
import { Link } from 'react-router-dom';
import { PokemonQuery_pokemon_v2_pokemon_pokemon_v2_pokemonabilities } from '../graphql/generated/PokemonQuery';

interface Props {
    id: number;
    name: string;
    height: number | null;
    weight: number | null;
    pokemon_v2_pokemonabilities: PokemonQuery_pokemon_v2_pokemon_pokemon_v2_pokemonabilities[];
}

const PokemonCard = ({ id = 5, name = 'Pokemon', weight, height, pokemon_v2_pokemonabilities: abilities }: Props) => {
    return (
        <PCard sx={{ maxWidth: 345 }}>
            <CardActionArea>
                <CardMedia
                    component="img"
                    height="250"
                    src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${id}.png`}
                    alt={name}
                />
                <CardContent>
                    <Typography gutterBottom variant="h4" component="div">
                        {capitalize(name)}
                    </Typography>
                    <Typography variant="body1" color="text.secondary" component="div">
                        Height:
                        <AttributeValue>{height}</AttributeValue>
                    </Typography>
                    <Typography variant="body1" color="text.secondary" component="div">
                        Weight:
                        <AttributeValue>{weight}</AttributeValue>
                    </Typography>
                    <Typography variant="body1" color="text.secondary" component="div">
                        Abilities:
                        <AttributeValue>{abilities.map((a) => a.pokemon_v2_ability?.name).join(', ')}</AttributeValue>
                    </Typography>
                </CardContent>
            </CardActionArea>
            <CardActions>
                <Button component={Link} to={'/' + id}>
                    More
                </Button>
            </CardActions>
        </PCard>
    );
};

const AttributeValue = styled(Box)({
    paddingLeft: 5,
    display: 'inline',
    fontWeight: 500,
});

const PCard = styled(Card)({
    minWidth: 200,
});

export default PokemonCard;
