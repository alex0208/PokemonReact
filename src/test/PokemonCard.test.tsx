import { cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import PokemonCard from '../components/PokemonCard';

afterEach(cleanup);

it('matches snapshot', () => {
    const props = {
        id: 5,
        name: 'charmeleon',
        height: 11,
        weight: 190,
    };

    const tree = renderer
        .create(
            <MemoryRouter>
                <PokemonCard {...props} />
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
