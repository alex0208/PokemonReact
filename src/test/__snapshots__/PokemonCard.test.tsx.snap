// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root-jqRKNJ kHBPTd MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root-haibrP BCqEU MuiCard-root sc-crrszt OERGE"
>
  <button
    className="MuiButtonBase-root-JTLGk xjigV MuiButtonBase-root MuiCardActionArea-root-dPTAie dResrh MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="charmeleon"
      className="MuiCardMedia-root-iQlcPv iIHDVI MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="250"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/5.png"
    />
    <div
      className="MuiCardContent-root-gAaESu jGCBVS MuiCardContent-root"
    >
      <div
        className="MuiTypography-root-bWuLxc dgkYgr MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
      >
        Charmeleon
      </div>
      <div
        className="MuiTypography-root-bWuLxc iTBnDU MuiTypography-root MuiTypography-body1"
      >
        Height:
        <div
          className="sc-bdfBQB sc-iqHYmW dMAMKp MuiBox-root"
        >
          11
        </div>
      </div>
      <div
        className="MuiTypography-root-bWuLxc iTBnDU MuiTypography-root MuiTypography-body1"
      >
        Weight:
        <div
          className="sc-bdfBQB sc-iqHYmW dMAMKp MuiBox-root"
        >
          190
        </div>
      </div>
      <div
        className="MuiTypography-root-bWuLxc iTBnDU MuiTypography-root MuiTypography-body1"
      >
        Abilities:
        <div
          className="sc-bdfBQB sc-iqHYmW dMAMKp MuiBox-root"
        >
          
        </div>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-jRDNLQ dtaseF MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root-iRhwuq cfBnLq MuiCardActions-root MuiCardActions-spacing"
  >
    <a
      className="MuiButtonBase-root-JTLGk xjigV MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiButton-root-gxQIQh HSLTP"
      href="/5"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      More
    </a>
  </div>
</div>
`;
